File: LeetCode Challenge/LeetCode Youtube/YuhLapTrinhVien/Vol(0-199)/1518.go
package main

import "fmt"

func numWaterBottles(numBottles int, numExchange int) int {
	totalDrunk := numBottles
	emptyBottles := numBottles

	for emptyBottles >= numExchange {
		newBottles := emptyBottles / numExchange
		emptyBottles = emptyBottles%numExchange + newBottles
		totalDrunk += newBottles
	}
	return totalDrunk
}

func main() {
	fmt.Println(numWaterBottles(9, 3))
	fmt.Println(numWaterBottles(15, 4))
}

File: LeetCode Challenge/LeetCode Youtube/YuhLapTrinhVien/Vol(0-199)/1518/1518.ts
function numWaterBottles(numBottles: number, numExchange: number): number {
  let totalDrunk: number = numBottles;
  let emptyBottles: number = numBottles;

  while (emptyBottles >= numExchange) {
    const newBottles: number = Math.floor(emptyBottles / numExchange);
    emptyBottles = (emptyBottles % numExchange) + newBottles;
    totalDrunk += newBottles;
  }
  return totalDrunk;
}

// Main function equivalent
function main(): void {
  console.log(numWaterBottles(9, 3)); // Output: 13
  console.log(numWaterBottles(15, 4)); // Output: 19
}

main();
